src/solver/conv_winoRxS_ultra_f2x3.cpp

```diff
@@ -0,0 +1,864 @@
+            return false;
+    }
+    const auto grid_workgroup_count_x = params.GetStream().GetMaxHardwareComputeUnits();
+    if(!params.IsLayoutDefault())
+    {
+        return false;
+    }
+
+    constexpr auto ELEM_SZ    = static_cast<int64_t>(sizeof(half_float::half));
+    constexpr auto D_W_PITCH  = ELEM_SZ * 1;
+    constexpr auto O_W_PITCH  = ELEM_SZ * 1;
+    const auto D_H_PITCH      = D_W_PITCH * W;
+    const auto O_H_PITCH      = O_W_PITCH * OW;
+    const auto D_C_PITCH      = D_H_PITCH * H;
+    const auto O_K_PITCH      = O_H_PITCH * OH;
+    const auto D_N_PITCH      = D_C_PITCH * C;
+    const auto O_N_PITCH      = O_K_PITCH * K;
+    const auto TILES_N_ROW    = (OW + o_tile_step_W - 1) / o_tile_step_W;
+    const auto TILES_N_COLUMN = (OH + o_tile_step_H - 1) / o_tile_step_H;
+
+    const auto D_STEP_1_PITCH = d_tile_step_H * D_H_PITCH - TILES_N_ROW * d_tile_step_W * D_W_PITCH;
+    const auto O_STEP_1_PITCH = o_tile_step_H * O_H_PITCH - TILES_N_ROW * o_tile_step_W * O_W_PITCH;
+    const auto D_STEP_2_PITCH = D_N_PITCH - TILES_N_COLUMN * d_tile_step_H * D_H_PITCH;
+    const auto O_STEP_2_PITCH = O_N_PITCH - TILES_N_COLUMN * o_tile_step_H * O_H_PITCH;
+
+    // clang-format off
+    return C <= 240
+        && K <= 16
+        && S <= 3
+        && R <= 3
+        && D_H_PITCH < std::pow(2, 16)
+        && O_H_PITCH < std::pow(2, 16)
+        && D_C_PITCH < std::pow(2, 30)
+        && O_K_PITCH < std::pow(2, 30)
+        && D_STEP_1_PITCH < std::pow(2, 18)
+        && O_STEP_1_PITCH < std::pow(2, 18)
+        && D_STEP_2_PITCH < std::pow(2, 30)
+        && O_STEP_2_PITCH < std::pow(2, 30)
+        && grid_workgroup_count_x < std::pow(2, 16);
+    // clang-format on
+}
+#endif
+
+template <typename T>
+void* CopyDataToSymbol(const Handle& handle,
+                       const Kernel& kernel,
+                       const std::vector<T> data,
+                       const std::string& name)
+{
+#if MIOPEN_BACKEND_HIP
+    const auto module    = kernel.program.GetModule();
+    const auto data_size = data.size() * sizeof(T);
+
+    size_t dev_buf_sz;
+    hipDeviceptr_t dev_buf_ptr;
+    auto status = hipModuleGetGlobal(&dev_buf_ptr, &dev_buf_sz, module, name.c_str());
+    if(status != hipSuccess)
+        MIOPEN_THROW_HIP_STATUS(status, "Failed hip to get control_buf info");
+
+    if(dev_buf_sz < data_size)
+        MIOPEN_THROW("Buffer size is than required");
+
+    handle.Copy(static_cast<const void*>(data.data()), dev_buf_ptr, data_size);
+
+    return static_cast<void*>(dev_buf_ptr);
+#else
+    (void)handle; // -warning
+    (void)kernel; // -warning
+    (void)data;   // -warning
+    (void)name;   // -warning
+    return nullptr;
+#endif
+}
+
+} // namespace
+
+PerformanceConfigConvBinWinogradUltraRxSf2x3::PerformanceConfigConvBinWinogradUltraRxSf2x3(
+    int intl_factor_)
+    : intl_factor(intl_factor_)
+{
+}
+
+void PerformanceConfigConvBinWinogradUltraRxSf2x3::HeuristicInit(const ConvolutionContext& config)
```

Comment:
Question: how important is tuning for performance? I.e. what if we just use HeuristicInit and don't do Tuning? How big performance drop shall we expect?

Reply:
For now performance model with `Interleave Factor` is not ready. We decided to disable tuning and set `Interleave Factor` to it's default value `1`. A performance model will be added in a subsequent PR.